src/stdatm/speed_parameters.py

```diff
@@ -10,3 +11,249 @@
+    return pressure * (COEFF_2 * mach**7 / (7 * mach**2 - 1.0) ** 2.5 - 1.0)
+
+
+@singledispatch
+def compute_impact_pressure(mach, pressure):
+    """
+
+    :param mach: no unit
+    :param pressure: in Pa
+    :return: impact pressure in Pa
+    """
+    # Implementation for numpy arrays
+    mach = np.asarray(mach)
+    idx_subsonic = mach <= 1.0
+    idx_supersonic = np.logical_not(idx_subsonic)
+
+    if np.shape(pressure) != np.shape(mach):
+        pressure = np.broadcast_to(pressure, np.shape(mach))
+    else:
+        pressure = np.asarray(pressure)
+
+    impact_pressure = np.empty_like(mach)
+    impact_pressure[idx_subsonic] = _compute_subsonic_impact_pressure(
+        mach[idx_subsonic], pressure[idx_subsonic]
+    )
+    impact_pressure[idx_supersonic] = _compute_supersonic_impact_pressure(
+        mach[idx_supersonic], pressure[idx_supersonic]
+    )
+    return impact_pressure
+
+
+@compute_impact_pressure.register
+def _(mach: Real, pressure: Real):
+    # Implementation for floats
+    if mach <= 1.0:
+        return _compute_subsonic_impact_pressure(mach, pressure)
+
+    return _compute_supersonic_impact_pressure(mach, pressure)
+
+
+# CALIBRATED AIRSPEED =================================================================
+#         Computation is done using Eq. 3.16 and 3.17 from:
+#         Gracey, William (1980), "Measurement of Aircraft Speed and Altitude",
+#         NASA Reference Publication 1046.
+#         https://apps.dtic.mil/sti/pdfs/ADA280006.pdf
+#         These formula allow to compute the impact pressure from CAS. The formula to be used
+#         is decided according to comparison between CAS value and speed of sound at sea level.
+#         Here we use them the other way around, which explains why the equation to be used is,
+#         oddly, decided by its result.
+
+# Pre-calculation of some equation constants for sake of speed.
+GAMMA_MINUS_ONE_OVER_GAMMA = (GAMMA - 1.0) / GAMMA
+GAMMA_MINUS_ONE_OVER_TWO_GAMMA = (GAMMA - 1.0) / (2.0 * GAMMA)
+COEFF_1 = 6.0**2.5 * 1.2**3.5
+
+
+def _compute_cas_low_speed(impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    # To be used when resulting CAS is lower than sea_level_speed_of_sound
+    return sea_level_speed_of_sound * np.sqrt(
+        5.0 * ((impact_pressure / sea_level_pressure + 1.0) ** GAMMA_MINUS_ONE_OVER_GAMMA - 1.0)
+    )
+
+
+def _equation_cas_high_speed(cas, impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    # To be used when resulting CAS is greater than sea_level_speed_of_sound
+    return (
+        cas
+        - sea_level_speed_of_sound
+        * (
+            (impact_pressure / sea_level_pressure + 1.0)
+            * (7.0 * (cas / sea_level_speed_of_sound) ** 2 - 1.0) ** 2.5
+            / COEFF_1
+        )
+        ** GAMMA_MINUS_ONE_OVER_TWO_GAMMA
+    )
+
+
+def _compute_cas_high_speed(impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    solution = root(
```

Comment:
I think you explained that to me but isn't it possible to use the single_dispatch to use either root or root_scalar depending on whether we have a float or an array ?

Reply:
The reason was that it was not efficient, but since, I changed the resolution method in 5b8d8144608408228c20ea82f596ca336b7c9c6f, and we can get a significant gain.